src/plugins/home/public/application/components/tutorial/instruction.js

```diff
@@ -36,6 +45,10 @@ export function Instruction({ commands, paramValues, textPost, textPre, replaceT
       </div>
     );
   }
+  let custom;
+  if (customComponent === 'apm_fleet') {
+    custom = <APMFleet />;
+  }
```

Comment:
Can't we have the `customComponent ` passed down be the actual react component, instead of passing down a string and hardcoding a special case here? I feel like this logic should be handled on an upper level.

Reply:
Unfortunately, that is not possible @pgayvallet. 

I need a component that is rendered every time the tutorial is opened. The way the tutorial works now is: I register a new tutorial on plugin setup in the server, and a static JSON is passed to it. 

The APM tutorial then must change based on some runtime information, so it must live in the Client and must be executed every time the page is rendered.

Besides refactoring all the tutorial registration way, I couldn't figure out another way of doing it (and refactoring is definitely out of scope of this issue).

I'm open for suggestions if you have any other idea.

Reply:
> The APM tutorial then must change based on some runtime information, so it must live in the Client and must be executed every time the page is rendered.

I was more suggesting to have this `customComponent` -> `component` mapping be performed on a higher level on the component tree of the `home` plugin. Maybe in `src/plugins/home/public/application/components/tutorial/tutorial.js` or in `src/plugins/home/public/application/load_tutorials.js`